src/Utils/BlockTemplateUtils.php

```diff
@@ -422,17 +422,21 @@ public static function theme_has_template_part( $template_name ) {
 	/**
 	 * Checks to see if they are using a compatible version of WP, or if not they have a compatible version of the Gutenberg plugin installed.
 	 *
+	 * @param  boolean $template_part Whether we are checking for a template part or not.
 	 * @return boolean
 	 */
-	public static function supports_block_templates() {
-		if (
-			! wc_current_theme_is_fse_theme() &&
-			( ! function_exists( 'gutenberg_supports_block_templates' ) || ! gutenberg_supports_block_templates() )
-		) {
-			return false;
+	public static function supports_block_templates( $template_part = false ) {
+		if ( (bool) $template_part && ( wc_current_theme_is_fse_theme() || current_theme_supports( 'block-template-parts' ) ) ) {
+				return true;
+		} else {
+			if (
+				! wc_current_theme_is_fse_theme() &&
+				( ! function_exists( 'gutenberg_supports_block_templates' ) || ! gutenberg_supports_block_templates() )
+			) {
+				return false;
+			}
+			return true;
```

Comment:
This logic is really ugly and could be cleaned up.

Reply:
I refactored this in 64b79d088b8faf28903c1f0e693409a30b8a6f8a.